pkgs/build-support/substitute/substitute-attrs.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, substitute }:
+
+/*
+  This is a wrapper around `substitute` in the stdenv, designed to be as easy to use as
+  `substituteAll` but have fewer pitfalls.
+
+  Attribute arguments:
+
+  - `srcOrAttrs`: either the path to the file to substitute, or an attrset that fits into `substitute`
```

Comment:
I think this can be just
```suggestion
  - `path`: A [string-like value](https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.strings.isStringLike) of a [store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path).
```

Then both `./foo.patch` and derivations can be passed. It's a fairly standard notion to do this.

Oh and I'd also either implement it already, or at least leave the possibility open, to support full directories as well, instead of just single files.

Reply:
I took Silvan's text.